pkg/sql/distsql_plan_stats.go

```diff
@@ -512,9 +512,54 @@ func constructUsingExtremesPredicate(
 ) string {
 	var predicate string
 	if index.GetKeyColumnDirection(0) == catpb.IndexColumn_ASC {
-		predicate = fmt.Sprintf("%s < %s OR %s > %s OR %s is NULL", columnName, extremesSpans.Get(0).EndKey().Value(0).String(), columnName, extremesSpans.Get(1).StartKey().Value(0).String(), columnName)
+		// Construct the comparison operators
+		lbExpr := tree.ComparisonExpr{
+			Operator: treecmp.MakeComparisonOperator(treecmp.LT),
+			Left:     &tree.ColumnItem{ColumnName: tree.Name(columnName)},
+			Right:    extremesSpans.Get(0).EndKey().Value(0),
```

Comment:
nit: Both `if/else` cases look identical except for the span values. You could simplify things by just setting the two span values to variable in `if/else` and moving the expression tree construction outside the `if/else`.

nit: Some decomposition of `constructUsingExtremesSpans` might simplify things further so you don't have to reach into spans here. You could have a function that returns the histogram upper and lower bound, and pass those to both `constructUsingExtremesSpans` and `constructUsingExtremesPredicate`.

Reply:
Fixed!